app/javascript/lib/metrics-api.js

```diff
@@ -330,6 +331,22 @@ function getAnalyticsPageName() {
   return getSCPContext().analyticsPageName
 }
 
+/**
+ * gets the tab name for analytics
+ */
+function getTabProperty() {
+  const url = window.location.toString()
+  if (url.includes('step=')) {
+    // if in an upload wizard tab
+    return url.replace(/^(.*?)?step=/, '').trim().replace(/\s/g, '-')
+  } else if (url.includes('#study')) {
+    // if in a study tab
+    return url.replace(/^(.*?)#/, '').trim().replace(/\s/g, '-')
```

Comment:
I hadn't considered this implementation route, but it strikes me as reasonable!  

It's a bit less directly connected to the UX (and thus a bit less intuitive for analysts like, say, a PM), but also likely more stable than using some normalized string version of the tab's displayed text.

It'd be nice if the tab's displayed text were also logged, but if that's substantial effort then maybe it's worth deferring.

Reply:
We can alter the display name in Mixpanel so someone like a PM can have the nice human styled names for these properties even if the raw data version is less nice

Reply:
By "the tab's displayed text", I mean the property value, not the property name.

Needing to mentally map the tab name to something that appears in the URL (rather than the main GUI) isn't especially intuitive, and I could see folks not making the connection and getting stuck.